src/components/landing/landing.js

```diff
@@ -1,52 +1,47 @@
-﻿import React, { Component } from "react";
+﻿import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import { connect } from "react-redux";
-import SliderForEvents from "./SliderForEvents/SliderForEvents";
-import EventsListItem from "./EventsListItem/EventsListItem";
 import "./landing.css";
 import { getUpcomingEvents } from "../../actions/event/event-list-action";
 import HeadArticle from "./HeadArticle";
-import EventMatrix from "./EventMatrix/EventMatrix";
 import landingConstants from "../../constants/landingConstants";
+import EventsViewMode from "./EventsViewMode/EventsViewMode";
+import viewModeSwitcher from "../helpers/landingUtils";
 
 const { UPCOMING_EVENTS } = landingConstants;
-class Landing extends Component {
-  componentDidMount() {
-    this.props.getUpcomingEvents();
-  }
 
-  handleClick = () => {
-    this.props.onSubmit();
-  };
+const Landing = ({ getUpcomingEventsDispatch, events }) => {
+  const [eventsViewMode, setEventsViewMode] = useState("slider");
+
+  useEffect(() => {
+    getUpcomingEventsDispatch();
+  }, [getUpcomingEventsDispatch]);
```

Comment:
It is not the same as `componentDidMount`

Reply:
Removed dependencies, left an empty array ✔